tools/check_release.py

```diff
@@ -0,0 +1,455 @@
+    cwd = os.getcwd()
+
+    repo = "https://" + user + '@developer.mbed.org/users/' + user + '/code/' + test 
+
+    # Clone the repo if it doesn't already exist
+    path = abspath(repo_path + '/' + test)
+    if not os.path.exists(path):
+        print("Test repo doesn't exist, cloning...")
+        os.chdir(abspath(repo_path))        
+        clone_cmd = ['hg', 'clone', repo]
+        run_cmd(clone_cmd, exit_on_failure=True)
+    
+    os.chdir(path)
+
+    client = hglib.open(path)        
+
+    lib_file = library + '.lib'    
+    if os.path.isfile(lib_file):
+        # Rename command will fail on some OS's if the target file already exist,
+        # so ensure if it does, it is deleted first.
+        bak_file = library + '_bak' 
+        if os.path.isfile(bak_file):
+            os.remove(bak_file)
+        
+        os.rename(lib_file, bak_file)
+    else:
+        print("!! Error trying to backup lib file prior to updating.")
+        return False
+    
+    # mbed 2 style lib file contains one line with the following format
+    # e.g. https://developer.mbed.org/users/<user>/code/mbed-dev/#156823d33999
+    exp = 'https://developer.mbed.org/users/' + user + '/code/' + library + '/#[A-Za-z0-9]+'
+    lib_re = re.compile(exp)
+    updated = False
+
+    # Scan through mbed-os.lib line by line, looking for lib version and update it if found
+    with open(bak_file, 'r') as ip, open(lib_file, 'w') as op:
+        for line in ip:
+
+            opline = line
+            
+            regexp = lib_re.match(line)
+            if regexp:
+                opline = 'https://developer.mbed.org/users/' + user + '/code/' + library + '/#' + ref
+                updated = True
+    
+            op.write(opline)
+
+    if updated:
+
+        # Setup the default commit message
+        commit_message = '"Updating ' + library + ' to ' + ref + '"' 
+ 
+        # Setup and run the commit command. Need to use the rawcommand in the hglib for this in order to pass
+        # the string value to the -m option. run_cmd using subprocess does not like this syntax.
+        try:
+            client.rawcommand(['commit','-m '+commit_message, lib_file])
+
+            cmd = ['hg', 'push', '-f', repo]
+            run_cmd(cmd, exit_on_failure=True)
+            
+        except:
+            print("Lib file already up to date and thus nothing to commit")
+                            
+    os.chdir(cwd)
+    return updated
+
+def get_sha(repo_path, library):
+    """ Gets the latest SHA for the library specified. The library is assumed to be
+        located at the repo_path. If a SHA cannot be obtained this script will exit.
+
+    Args:
+    library - library name
+    repo_path - path to the user's repo area
+    
+    Returns:
+    sha - last commit SHA
+    """
+    cwd = os.getcwd()
+    sha = None
+    os.chdir(abspath(repo_path + '/' + library))
+    
+    cmd = ['hg', 'log', '-l', '1']
+    ret, output = run_cmd_with_output(cmd, exit_on_failure=True)
```

Comment:
I agree with @bridadan on this one, I find it much clearer to read the code with the intermediate variable (in this case).